### PR TITLE
improved actionchain error handling

### PR DESCRIPTION
**Pull request by lucidd**
## What does this PR change?

This PR adds error handling when the actionchain module `mgractionchains` is not present on a minion. Before this would not complete the actions and only a later cleanup task would set them to failed. Now those actions will be set to failed directly with the proper error message.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/12826

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql" 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/900-actionchaindispatch.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE rhnActionChain DROP CONSTRAINT IF EXISTS rhn_actionchain_dispatched_ck;
```
Comment (mcalmer) [inline schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/900-actionchaindispatch.sql:1]: Head got tagged today. So new schema migration dir is 4.2.5-to-4.2.6

**Review by mcalmer — CHANGES_REQUESTED** [review]
The schema migration needs to move as uyuni was tagged today.

File: java/code/src/com/suse/manager/webui/websocket/VirtNotifications.java [hunk 2]
```diff
@@ -139,8 +139,7 @@ public void onMessage(Session session, String messageBody) {
                     // Send out a list of the latest pending actions to display
                     Server server = SystemManager.lookupByIdAndUser(newId, userOpt.get());
                     List<ServerAction> serverActions = ActionFactory.listServerActionsForServer(server,
-                            Arrays.asList(ActionFactory.STATUS_QUEUED,
-                                          ActionFactory.STATUS_PICKEDUP,
+                            Arrays.asList(ActionFactory.STATUS_QUEUED, ActionFactory.STATUS_PICKED_UP,
                                           ActionFactory.STATUS_PICKED_UP));
```
Comment (meaksh) [inline java/code/src/com/suse/manager/webui/websocket/VirtNotifications.java:143]: It seems we're duplicating `ActionFactory.STATUS_PICKED_UP` here.

IIUC this shoud be:
 ```suggestion
                             Arrays.asList(ActionFactory.STATUS_QUEUED,
                                           ActionFactory.STATUS_PICKED_UP));
```

**Review by meaksh — COMMENTED** [review]
Besides of what @mcalmer already mentioned about the schema, I've just spotted a minor thing to fix.

Other than that, let's give this a try :+1:

**Review by mbologna — APPROVED** [review]

File: schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/900-actionchaindispatch.sql (hunk 1 above)
Comment (mcalmer) [reply schema/spacewalk/upgrade/susemanager-schema-4.2.4-to-susemanager-schema-4.2.5/900-actionchaindispatch.sql:1]: JFYI - we just had to make another schema release. So just lookup the correct directory when the branch is open again to merge :-)

File: java/code/src/com/suse/manager/webui/websocket/VirtNotifications.java (hunk 2 above)
Comment (meaksh) [reply java/code/src/com/suse/manager/webui/websocket/VirtNotifications.java:143]: Hmmm, was this change missed after the force push?

**Review by mcalmer — APPROVED** [review]
I think we are good to go

**Pull request merged**